### PR TITLE
Remove experimental code

### PR DESCRIPTION
**Pull request by AspirinSJL**
This PR reverts the changes in #11433 that are beyond reordering.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by yang-g** [comment]
Does it work?

**Comment by AspirinSJL** [comment]
@yang-g I ran `tools/run_tests/run_tests.py -l c++ -r "grpclb_end" -n inf` for ~20 mins, the flakiness increases from 0.13% to 0.18% (without new flakes) after this PR.

File: src/core/lib/surface/init.c [hunk 1]
```diff
@@ -180,8 +179,6 @@ void grpc_shutdown(void) {
       GRPC_EXEC_CTX_INITIALIZER(0, grpc_never_ready_to_finish, NULL);
   gpr_mu_lock(&g_init_mu);
   if (--g_initializations == 0) {
-    grpc_executor_shutdown(&exec_ctx);
-    grpc_timer_manager_set_threading(false);  // shutdown timer_manager thread
```
Comment (sreecha) [inline src/core/lib/surface/init.c]: This should stay. I do not see `grpc_timer_manager_set_threading(false)` being called from anyplace else.

Comment (yang-g) [reply src/core/lib/surface/init.c]: I think the effect of this change is to shutdown the executor and the timer_manager threads within grpc_iomgr_shutdown, which is after the plugin shutdowns.

**Comment by yang-g** [comment]
@AspirinSJL Thanks. How about other (non grpclb) tests?

**Comment by AspirinSJL** [comment]
@yang-g I haven't run other tests. Should I run some or all?

**Comment by yang-g** [comment]
@AspirinSJL These two lines were added because they seemed to reduce test failure rates. I guess looking at all the tests to see whether any new test failures pop up is a good idea.

**Comment by AspirinSJL** [comment]
@yang-g Yeah, that makes sense. Running `tools/run_tests/run_tests.py -l c++`.

**Comment by AspirinSJL** [comment]
@yang-g I've also submitted an internal global test for this change.

**Comment by yang-g** [comment]
Please also run with `-l c`

File: src/core/lib/surface/init.c (hunk 1 above)
Comment (sreecha) [reply src/core/lib/surface/init.c]: Yeah, but `grpc_timer_manager_set_threading` is not being called in `grpc_iomgr_shutdown`.  

So, I think we should add `grpc_timer_manager_set_threading` back in `grpc_iomgr_shutdown`

Comment (yang-g) [reply src/core/lib/surface/init.c]: You are right. Maybe moving the call to around `grpc_iomgr_shutdown` in here?

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by AspirinSJL** [comment]
Already removed.

**Pull request closed without merging**